test/test_core.py

```diff
@@ -7722,6 +7722,41 @@ def test_source_map(self, args):
     else:
       self.assertTrue(seen_lines.issuperset([6, 7, 11, 12]), seen_lines)
 
+  @no_asan("asan does not work with dynamic linking")
+  @needs_dylink
+  def test_embind_dylink_visibility_hidden(self):
+    # Check that embind is usable from a library built with "-fvisibility=hidden"
+    create_file('liba.cpp', r'''
+      #include <emscripten/val.h>
+      #define EXPORT __attribute__((visibility("default")))
+      using namespace emscripten;
+      EXPORT void liba_fun() {
+        unsigned char buffer[1];
+        val view(typed_memory_view(1, buffer));
+      }
+    ''')
+
+    # side module
+    self.build_dlfcn_lib('liba.cpp', outfile='liba.so', emcc_args=['-std=c++11', '-fvisibility=hidden'])
```

Comment:
Most other tests that call this function use `self.prep_dlfcn_main()` after.  Is there some reason not to do that here?

Reply:
It sets NO_AUTOLOAD_DYLIBS, which causes an assertion "Assertion failed: undefined symbol '_ZTIN10emscripten11memory_viewIhEE'. "

Other than that it's fine, so we can for instance use it and then clear NO_AUTOLOAD_DYLIBS.